src/Microsoft.AspNetCore.Mvc.Razor/Internal/CompilerCache.cs

```diff
@@ -67,27 +79,26 @@ public CompilerCache(IFileProvider fileProvider)
         /// <inheritdoc />
         public CompilerCacheResult GetOrAdd(
             string relativePath,
-            Func<string, CompilerCacheContext> cacheContextFactory)
+            Func<string, CompilationResult> compile)
```

Comment:
If this ends up being something like:
```
public interface IViewCompiler
{
    public Task<CompiledViewDescriptor> CompileAsync(string relativePath)
}
```

It's way simpler, and the fact that we have compilation at all is just a detail.